src/commands/auth/device/login.ts

```diff
@@ -52,6 +52,14 @@ export default class Login extends SfdxCommand {
   public async run(): Promise<AuthFields> {
     if (await Prompts.shouldExitCommand(this.ux, this.flags.noprompt)) return {};
 
+    if (this.flags.instanceurl) {
+      const instanceurl = this.flags.instanceurl as string;
```

Comment:
All of these should be moved to [Common.resolveLoginUrl](https://github.com/salesforcecli/plugin-auth/blob/main/src/common.ts#L45) *after* we get a final URL. i.e. if `instanceUrl` isn't passed in, we then get it from `sfdx-project.json`, *then* we should check if it is valid. The message should be able to be defined in one file.